src/StreamJsonRpc/Reflection/MessageFormatterRpcMarshaledContextTracker.cs

```diff
@@ -87,6 +87,44 @@ private enum MarshalMode
             MarshallingRealObject = 1,
         }
 
+        internal static (JsonRpcProxyOptions ProxyOptions, JsonRpcTargetOptions TargetOptions)? GetMarshalOptionsForType(Type type)
+        {
+            Requires.NotNull(type, nameof(type));
+
+            foreach ((Type implicitlyMarshaledType, JsonRpcProxyOptions proxyOptions, JsonRpcTargetOptions targetOptions) in ImplicitlyMarshaledTypes)
+            {
+                if (implicitlyMarshaledType == type ||
+                    (implicitlyMarshaledType.IsGenericTypeDefinition &&
+                     type.IsConstructedGenericType &&
+                     implicitlyMarshaledType == type.GetGenericTypeDefinition()))
+                {
+                    return (proxyOptions, targetOptions);
+                }
+            }
+
+            if (type.IsInterface && type.GetCustomAttribute<RpcMarshalableAttribute>() is not null)
+            {
+                if (typeof(IDisposable).IsAssignableFrom(type) is false)
+                {
+                    throw new NotSupportedException(Resources.MarshalableInterfaceNotDisposable);
+                }
+
+                if (type.GetEvents().Any())
```

Comment:
using the `.Any()` extension method is slower than using the `Length` or `Count` properties, and sometimes allocates memory when the property alternatives would not. For each of these, as they return arrays can you use the `Length` property instead?